apps/language_server/lib/language_server/providers/formatting.ex

```diff
@@ -50,7 +50,8 @@ defmodule ElixirLS.LanguageServer.Providers.Formatting do
   end
 
   def should_format?(file_uri, project_dir, inputs) when is_list(inputs) do
-    file = String.trim_leading(file_uri, "file://")
+    project_dir = project_dir |> String.downcase()
+    file_path = file_uri |> SourceFile.path_from_uri() |> String.downcase()
```

Comment:
I'm afraid it won't work on case-sensitive file systems

Reply:
Hi @lukaszsamson , this is exactly what's happening going on in `can_format?` a few lines above. (I held off the refactor)
If there is a better way, I've happy to improve the PR and fix `can_format?` as well.

Reply:
That code has been there for 2 years, so I'm guessing it's fine? 🤷‍♂️

Reply:
@princemaple
I also think that this won't work for case-sensitive file systems.

Reply:
> That code has been there for 2 years, so I'm guessing it's fine? 🤷‍♂️

Maybe it was broken the whole time. I can't check it right now, I usually work on CI filesystems (macOS and Windows).
Anyway it seems the only reliable way of determining if the file system is CI or CS is by creating/enumerating files, see e.g. https://stackoverflow.com/questions/430256/how-do-i-determine-whether-the-filesystem-is-case-sensitive-in-net/430810

Reply:
> I usually work on CI filesystems (macOS and Windows)

OK, same here.

Reply:
I've come to the conclusion that the bug is that `Path.wildcard` returns `c:/` in lowercase.